pkg/controller/registry/reconciler/reconciler.go

```diff
@@ -275,6 +275,17 @@ func Pod(source *operatorsv1alpha1.CatalogSource, name, opmImg, utilImage, img s
 				Name:      "catalog-content",
 				MountPath: catalogPath,
 			}
+			// init container to copy catalog info.
+			// ExtractContent.CatalogDir is mandatory when ExtractContent is provided
+			// ExtractContent.CacheDir is optional, so we only add it if it is set
+			var extractArgs = []string{
+				"--catalog.from=" + grpcPodConfig.ExtractContent.CatalogDir,
+				"--catalog.to=" + fmt.Sprintf("%s/catalog", catalogPath),
```

Comment:
Do we already have a check for this?

If ExtractContent is not empty and ExtractContent.CatalogDir is empty, we should raise an error with the following message:

"CatalogDir must be set when ExtractContent is provided."

Reply:
We do have that because CatalogDir is [listed as required](https://github.com/operator-framework/operator-lifecycle-manager/pull/3556/files#diff-cfdc2cf5582e0f68ab2efd954f992741d2d3053a5358c5a106ec64a3ab5d3826R978)